src/sparsezoo/objects/model_folder_objects.py

```diff
@@ -0,0 +1,442 @@
+                error
+        :param integration: name of the specific integration
+            (e.g. transformers, YOLOv5 etc.)
+        :return: boolean flag; True if File instance is loadable, otherwise False
+        """
+        _, extension = os.path.splitext(self.path)
+
+        if extension in self.loadable_extensions.keys():
+            validation_function = self.loadable_extensions[extension]
+            validation_function(strict_mode=strict_mode)
+            return True
+
+        else:
+            return False
+
+    def _validate_numpy(self, strict_mode):
+        if not load_numpy_list(self.path):
+            self._throw_error(
+                error_msg="Numpy file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_onnx(self, strict_mode):
+        if not onnx.load(self.path):
+            self._throw_error(
+                error_msg="Onnx file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_markdown(self, strict_mode):
+        try:
+            with open(self.path) as file:
+                file.readlines()
+        except Exception as error:  # noqa: F841
+            self._throw_error(
+                error_msg="Markdown file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_json(self, strict_mode):
+        try:
+            with open(self.path) as file:
+                json.load(file)
+        except Exception as error:  # noqa: F841
+            self._throw_error(
+                error_msg="Json file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_csv(self, strict_mode):
+        try:
+            with open(self.path) as file:
+                file.readlines()
+        except Exception as error:  # noqa: F841
+            self._throw_error(
+                error_msg="Csv file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_img(self, strict_mode):
+        if not Image.open(self.path):
+            self._throw_error(
+                error_msg="Image file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _throw_error(self, error_msg, strict_mode):
+        if strict_mode:
+            raise ValueError(error_msg)
+        else:
+            logging.warning(error_msg)
+
+
+class Directory(File):
+    """
+    Object that represents a directory.
+
+    :param files: list of files contained within the Directory
+    :param name: name of the Directory
+    :param path: path of the Directory
+    :param url: url of the Directory
+    """
+
+    def __init__(
+        self,
+        name: str,
+        files: Optional[List[File]] = None,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+
+        self.files = files
```

Comment:
can we automatically get the names of the files if given a local directory/tarball?

Reply:
Adding it to the new PR in the other branch.